### ISSUE TITLE
RangeError: Invalid time value

### ISSUE THREAD

**Issue by CoderCharlie** [date-fns/tz#8]
```
const now = new Date()
const nowTZ = TZDate.tz("America/New_York", now)
console.log(nowTZ.toString())
```

I changed it to a few other examples, still got the same error as below
```
RangeError: Invalid time value
```

**Comment by kossnocorp** [comment]
Hey! There's not enough info to do something with it. The environment, the version, your time zone, and so on.

![image](https://github.com/user-attachments/assets/d77cc6c4-1453-4b6c-aa15-a7932b79709a)

**Comment by CoderCharlie** [comment]
```
{
  "@date-fns/tz": "^1.0.2",
  "date-fns": "^4.1.0",
  "react-native": "0.73.4"
  ...
}
```
I was running a react-native app on my iPhone 15 simulator with iOS 17.4 on  MacOS 14.3.1 (23D60) with 2018 Macbook Pro

**Comment by kossnocorp** [comment]
@CoderCharlie, can you please check what this code returns in your environment:

```
new Intl.DateTimeFormat("en-GB", { timeZone: "Asia/Singapore", hour: "numeric", timeZoneName: "longOffset" }).format(new Date())
```

I don't have access to the simulator and React Native to test, but @kevin-abiera might be right because of the limited Intl support.

@kevin-abiera I guess there must be a way, but what you suggested doesn't work without `hour` and `timeZoneName`. I don't mind supporting the polyfill, but where do I look to see what is actually supported? I would appreciate a PR.

**Comment by kossnocorp** [comment]
Thank you, @kevin-abiera! I shipped the fix with `@date-fns/tz@1.1.2`
